go/vt/topo/stats_conn.go

```diff
@@ -160,6 +160,23 @@ func (st *StatsConn) Lock(ctx context.Context, dirPath, contents string) (LockDe
 	return res, err
 }
 
+// TryLock implements the Conn interface
+// As of today it provides same functionality as Lock
+func (st *StatsConn) TryLock(ctx context.Context, dirPath, contents string) (LockDescriptor, error) {
+	statsKey := []string{"Lock", st.cell}
+	if st.readOnly {
+		return nil, vterrors.Errorf(vtrpc.Code_READ_ONLY, readOnlyErrorStrFormat, statsKey[0], dirPath)
+	}
+	startTime := time.Now()
+	defer topoStatsConnTimings.Record(statsKey, startTime)
+	res, err := st.conn.TryLock(ctx, dirPath, contents)
+	if err != nil {
+		topoStatsConnErrors.Add(statsKey, int64(1))
+		return res, err
+	}
+	return res, err
```

Comment:
Why not just call Lock() here? This duplication could easily be missed in the future so that the TryLock actually does NOT match Lock.

Reply:
good point. I have changed it .